src/track/serato/markers2.cpp

```diff
@@ -679,6 +679,22 @@ RgbColor::optional_t SeratoMarkers2::getTrackColor() const {
     return std::nullopt;
 }
 
+void SeratoMarkers2::setTrackColor(RgbColor color) {
+    for (auto& pEntry : m_entries) {
+        DEBUG_ASSERT(pEntry);
+        if (pEntry->typeId() != SeratoMarkers2Entry::TypeId::Color) {
+            continue;
+        }
+        SeratoMarkers2ColorEntry* pColorEntry =
+                static_cast<SeratoMarkers2ColorEntry*>(pEntry.get());
+        pColorEntry->setColor(color);
+        return;
+    }
+
+    SeratoMarkers2ColorEntry* pEntry = new SeratoMarkers2ColorEntry(color);
```

Comment:
We should use `make_shared` here, maybe followed by a static_pointer_cast if necessary. Maybe an *upcast* of `shared_ptr` does not require explicit casting?

Reply:
Oh right, good idea.